lib/fictium/exporters/api_blueprint_exporter.rb

```diff
@@ -0,0 +1,59 @@
+require_relative 'api_blueprint_exporter/header_formatter'
+require_relative 'api_blueprint_exporter/resource_formatter'
+require_relative 'api_blueprint_exporter/action_formatter'
+require_relative 'api_blueprint_exporter/example_formatter'
+require_relative 'api_blueprint_exporter/footer_formatter'
+
+module Fictium
+  class ApiBlueprintExporter
```

Comment:
```rb
module Fictium
  module Exporters
    class ApiBlueprintExporter < BaseExporter
```

```rb
# lib/fictium/exporters/base_exporter.rb

module Fictium
  module Exporters
    class BaseExporter
      protected

      def build_header(document)
        raise 'SubclassResponsability'
      end

      def build_resources(document)
        raise 'SubclassResponsability'
      end

      def build_footer(document)
        raise 'SubclassResponsability'
      end

      def add_empty_lines # or another name
          select(&:present?).join("\n\n")
      end
    end
  end
end
```

Reply:
Each formatter is different, probably won't apply as much.